spec/controllers/dashboard_controller_spec.rb

```diff
@@ -90,6 +90,46 @@
     end
   end
 
+  context "external auth support" do
+    before { EvmSpecHelper.create_guid_miq_server_zone }
+
+    let(:user) { FactoryBot.create(:user, :role => "random") }
+    let(:user_name) { user.userid }
+    let(:user_domain) { "ipa.example.org" }
+    let(:user_email) { "#{user_name}@#{user_domain}" }
+    let(:user_group) { user.miq_groups.first.description }
+
+    it "initiates a task when logging in" do
+      stub_settings_merge(:authentication => {:mode => 'httpd', :httpd_role => true})
+      request.headers.merge!(
+        "HTTP_X_REMOTE_USER"        => user_name,
+        "HTTP_X_REMOTE_USER_EMAIL"  => user_email,
+        "HTTP_X_REMOTE_USER_DOMAIN" => user_domain,
+        "HTTP_X_REMOTE_USER_GROUPS" => user_group
+      )
+      skip_data_checks
+      post :external_authenticate, :params => {:user_name => user_name, :user_password => 'dummy'}
+      expect_failed_login # Not really failed, but waiting to complete
+
+      # In Rails test environment, we use memory session storage, but in production
+      # we use memcached. memcached Marshal.dump's the session, so we do that here
+      # manually to ensure it doesn't blow up.
+      Marshal.load(Marshal.dump(session))
```

Comment:
Am I correct that this is where it fails if you revert the prior change?

Reply:
Yes that's correct

Reply:
This seems contrived.  I wonder if there's a better way to test that.  Maybe one test for wait for task and one to test dump/load of session works?

Reply:
I mean it's all contrived haha.  Not sure what an alternative would be though?  Maybe we can discuss offline.